pkg/protocol/engine/engine.go

```diff
@@ -556,12 +532,94 @@ func (e *Engine) ErrorHandler(componentName string) func(error) {
 	}
 }
 
-func (e *Engine) RLock() {
-	e.accessMutex.RLock()
+func (e *Engine) initRootCommitment() {
```

Comment:
the function `EarliestRootCommitment` can be removed as this functionality is covered in here. Please also take the comment and put it here

Reply:
removed it